src/scmrepo/git/backend/dulwich/__init__.py

```diff
@@ -199,6 +200,8 @@ def clone(
             else:
                 repo = clone_from()
             cls._set_default_tracking_branch(repo)
+        except HTTPUnauthorized as exc:
+            raise AuthError(url) from exc
```

Comment:
For the clone, since we already have `CloneError`, it might be better to reuse that and only change the exception message.

Reply:
Yep, that's what I was thinking